### PR TITLE
update fe onboarding

### PR DESCRIPTION
**Pull request by SabrinaTardio**
Task/Issue URL: https://app.asana.com/0/0/1208077416568652/f

**Description**: Adds modification to the import dialog (reviewed in another PR), updates to use the new front end onboarding, re-set up the experiment

**Steps to test this PR**:
- Run ./clean-app.sh debug
- Comment in MainWindowController lines 73-83 and 85
- You should be assigned a cohort (oc or od)

Old onboarding:
- Smoke test
- Check that at the end no contextual onboarding is shown

New Onboarding
- When you get the new onboarding 
- You should see the new linear onboarding
- Test the actions work
- At every step check the correct step pixel is sent
- Check that at the end you see the contextual onboarding

Feature Flag
- Modify the flagger to return false
- Run ./clean-app.sh debug
- when the new onboarding is shown check the contextual onboarding is not shown
- modify the feature flagger back
- check that the contextual dialog is not shown

<!—
Tagging instructions
If this PR isn't ready to be merged for whatever reason it should be marked with the `DO NOT MERGE` label (particularly if it's a draft)
If it's pending Product Review/PFR, please add the `Pending Product Review` label.

If at any point it isn't actively being worked on/ready for review/otherwise moving forward (besides the above PR/PFR exception) strongly consider closing it (or not opening it in the first place). If you decide not to close it, make sure it's labelled to make it clear the PRs state and comment with more information.
—>

**Definition of Done**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

—
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)

### CODE REVIEW

File: DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -75,30 +57,12 @@
...
       "identity" : "lottie-spm",
       "kind" : "remoteSourceControl",
-      "location" : "https://github.com/airbnb/lottie-spm.git",
+      "location" : "https://github.com/airbnb/lottie-spm",
       "state" : {
         "revision" : "1d29eccc24cc8b75bff9f6804155112c0ffc9605",
         "version" : "4.4.3"
       }
     },
-    {
```
Comment (alessandroboron) [inline DuckDuckGo.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:84]: is this right?

File: DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift [hunk 2]
```diff
@@ -162,7 +162,11 @@ final class ContextualOnboardingStateMachine: ContextualOnboardingDialogTypeProv
 
     var state: ContextualOnboardingState {
         get {
-            return ContextualOnboardingState(rawValue: stateString) ?? .onboardingCompleted
+            if NSApp.delegateTyped.featureFlagger.isFeatureOn(.contextualOnboarding) {
```
Comment (alessandroboron) [inline DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift:165]: maybe you need to inject the feature flagger to control it in tests?

File: DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift [hunk 3]
```diff
@@ -381,4 +382,8 @@ final class ContextualOnboardingStateMachine: ContextualOnboardingDialogTypeProv
             break
         }
     }
+
+    func featureIsOff() {
```
Comment (alessandroboron) [inline DuckDuckGo/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachine.swift:386]:
```suggestion
    func turnOffFeature() {
```

File: UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift [hunk 4]
```diff
@@ -56,9 +58,20 @@ final class BrowserTabViewControllerOnboardingTests: XCTestCase {
...
         cancellables = []
         expectation = nil
+        featureFlagger = nil
         try super.tearDownWithError()
     }
 
+    func testWhenNavigationCompletedAndFeatureIsOffThenTurnOffFeature() throws {
+        featureFlagger.isFeatureOn = false
+        let expectation = self.expectation(description: "Wait for turnOffFeatureCalled to be called")
+        dialogProvider.dialogTypeForTabExpectation = expectation
```
Comment (alessandroboron) [inline UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift:68]: you meant to use `turnOffFeatureCalledExpectation` for dialogProvider?

Comment (SabrinaTardio) [reply UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift:68]: yes cause here conforms to both the dialogProvider and stateUpdater protocols

Comment (SabrinaTardio) [reply UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift:68]: sorry I misread yes you are right…

**Review by alessandroboron — APPROVED** [review]
Works as expected! Good Job!

**Pull request merged**